src/quarot/hf_utils.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+import pathlib
+
+import torch
+from transformers import AutoTokenizer, PreTrainedTokenizerBase
+
+from slicegpt.hf_utils import do_not_initialize
+
+from .model_adapter import ModelAdapter
+
+
+@do_not_initialize
+def get_model_and_tokenizer(
+    model_name: str,
+    model_path: str | None = None,
+    *,
+    uninitialized: bool = False,
+    dtype: torch.dtype = torch.float16,
+    token: str | bool | None = None,
+) -> tuple[ModelAdapter, PreTrainedTokenizerBase]:
+    """
+    Load the model and the tokenizer from the given path.
+    Set uninitialized to True when loading a pre-rotated and sliced model; in this case no weights are loaded
+    in this method.
+    The corresponding model adapter class must be imported before calling this method.
+    Scenarios:
+    - Rotate & slice HF model: model_name = name, model_path = empty, uninitialized = False
```

Comment:
These comments are probably not right, so we can replace with quarot specific (brief) comments.

Reply:
Agreed - this file should actually be deleted though once the adapter refactoring happens.